### ISSUE TITLE
[Feature] Store shared library name in tf.sysconfig

### ISSUE THREAD

**Issue by seanpmorgan** [tensorflow/tensorflow#27430]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 1.13.1 or 2.0.0
- Are you willing to contribute it (Yes/No): Probably not the best person

**Describe the feature and the current behavior/state.**
As of #22797 the tensorflow shared library is now versioned. This makes linking to the shared library difficult (at least with my knowledge of bazel). The recommended method in https://github.com/tensorflow/custom-op links a [static file name](https://github.com/tensorflow/custom-op/blob/master/tf/BUILD.tpl#L12)

**Will this change the current api? How?**
Add a new shared library name to tf.sysconfig 

**Who will benefit with this feature?**
The community who is building upon TensorFlow. Given the push for modularity it's an important thing to figure out.

**Any Other info.**
This stems from https://github.com/tensorflow/addons/pull/130

**Comment by seanpmorgan** [comment]
cc @yifeif as a change will need to be made to custom-op

**Comment by seanpmorgan** [comment]
As discussed in today's SIG Build ... a more short term fix (that wouldn't require API review) would be if there is a proper symlink from `libtensorflow_framework` to whatever the current version is.

**Comment by angersson** [comment]
@perfinion I think some internal tests may be broken due to the missing symlink:

```
==================== Test output for //tensorflow/tools/lib_package:libtensorflow_test:
+ CC=
+ TAR=
+ '[' -z '' ']'
+ CC=/usr/bin/gcc
+ '[' -z '' ']'
+ TAR=tar
+ TARFILE=/root/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/tools/lib_package/libtensorflow_test.runfiles/org_tensorflow/tensorflow/tools/lib_package/libtensorflow.tar.gz
+ CFILE=/root/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/tools/lib_package/libtensorflow_test.runfiles/org_tensorflow/tensorflow/tools/lib_package/libtensorflow_test.c
+ cd /root/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/_tmp/3302b1ea63f567e726bad68a4f03cae9
+ mkdir tensorflow
+ tar -xzf /root/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/tools/lib_package/libtensorflow_test.runfiles/org_tensorflow/tensorflow/tools/lib_package/libtensorflow.tar.gz -Ctensorflow
+ /usr/bin/gcc /root/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/tools/lib_package/libtensorflow_test.runfiles/org_tensorflow/tensorflow/tools/lib_package/libtensorflow_test.c -Itensorflow/include -Ltensorflow/lib -ltensorflow_framework -ltensorflow -oa.out
/usr/bin/ld: cannot find -ltensorflow_framework
collect2: error: ld returned 1 exit status
```

**Comment by perfinion** [comment]
I opened https://github.com/tensorflow/tensorflow/pull/27493 to add back the unversioned .so first.
That should fix this quickly then we can take a look at adding something to tf.sysconfig more carefully.

**Comment by perfinion** [comment]
@seanpmorgan this should work for you right? https://github.com/tensorflow/tensorflow/pull/27694

**Comment by seanpmorgan** [comment]
Yes that will work for us... we've already swapped to linking on versioned soname

**Comment by ageron** [comment]
I'm getting an error when trying to build TF Addons on MacOS:
```
ERROR: /private/var/tmp/.../external/local_config_tf/BUILD:3800:1: Executing genrule
@local_config_tf//:libtensorflow_framework.so.2 failed (Exit 1) bash failed: error
executing command /bin/bash -c ... (remaining 1 argument(s) skipped)
```

The `libtensorflow_framework.so.2` file is missing?

Looking into that directory, I see that the library's extension is dylib. So maybe the hardcoded name should be depend on the platform?

```bash
$ ls /Users/.../lib/python3.6/site-packages/tensorflow/
__init__.py				compiler				libtensorflow_framework.2.0.0.dylib	python
__pycache__				core					libtensorflow_framework.2.dylib		tools
_api					include					lite
```

**Comment by mihaimaruseac** [comment]
Yes, on macos they're `.dylib`s, not `.so`s

**Comment by angersson** [comment]
@ageron Thanks! See #27865

**Comment by seanpmorgan** [comment]
Solved by #27694 and #27865 ... Thanks for the help!
